ddl/ddl.go

```diff
@@ -466,13 +472,47 @@ func (d *ddl) newDeleteRangeManager(mock bool) delRangeManager {
 	return delRangeMgr
 }
 
+func (d *ddl) readyForConcurrencyDDL() {
+	workerFactor := func(tp workerType) func() (pools.Resource, error) {
+		return func() (pools.Resource, error) {
+			wk := newWorker(d.ctx, tp, d.sessPool, d.delRangeMgr, d.ddlCtx, true)
+			sessForJob, err := d.sessPool.get()
+			if err != nil {
+				return nil, err
+			}
+			sessForJob.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlmostFull)
+			wk.sess = newSession(sessForJob)
+			metrics.DDLCounter.WithLabelValues(fmt.Sprintf("%s_%s", metrics.CreateDDL, wk.String())).Inc()
+			return wk, nil
+		}
+	}
+	d.reorgWorkerPool = newDDLWorkerPool(pools.NewResourcePool(workerFactor(addIdxWorker), batchAddingJobs, batchAddingJobs, 3*time.Minute), reorg)
```

Comment:
I think you are unnecessary to set this timeout. the worker in the ```resPool``` are always available and we have limited the number of goroutine in the reorg stage.